MLSAKIET0821/MLSAKIET_GENAI_INTERNSHIP
# 2300290120271_vaishnavi_cs_2

I'm Vaishnavi Agrahari (2300290120271) ,from CS branch 2nd year and my domain is Generative AI.

My project is to "**Create a comprehensive NLP solution**".

This pull request introduces integration with Azure's Text Analytics API to perform comprehensive multi-language text analysis. It has been coded to process and analyze text files from a specified directory, including the **"Reviews"** folder, providing information on the language, sentiment, key phrases, entities, and linked entities in every document.

**Key Features and Improvement:**

**1. Azure Client Initialization:**
It utilizes **"AzureKeyCredential"** to provide secure connectivity to the Azure Text Analytics API.
It securely retrieves the endpoint and API key from environment variables, following best practice for credential management.

**2. Language Detection:**
The function detects the primary language of all text files placed in the folder **"Reviews"** and uses **"detect_language"** to determine the linguistic context for further processing.

**3. Sentiment Analysis:**
Implements **"analyze_sentiment"** to determine the overall sentiment, positive, neutral, or negative within each text file.

**4. Extract Key Phrase:**
- It extracts key phrases from the text using **"extract_key_phrases"** to highlight the most important ideas and topics.

**5. Entity Identification:**
- It identifies entities in text using **"recognize_entities"**, which categorizes them; for instance, location, organization, person, as a way of better comprehending the content.

**6. Linked Entity Recognition:**
It uses **"recognize_linked_entities"** to find and link entities with suitable URLs, as well as background information on the topic itself.

**Implementation Includes:**
- Environment configuration: Loads environment variables from a **".env"** file using **"python-dotenv"** for endpoint and API key configuration.
- File Processing Read all text files in reviews directory so that a huge amount of text documents can be processed in bulk.

- Error Handling: Includes a try-except block to capture and print any exceptions during execution, ensuring robust and fail-safe processing.

**How to Test:** 
1. Set up the **".env"** file with **"AI_SERVICE_ENDPOINT"** and **"AI_SERVICE_KEY"** for secure access to Azure Text Analytics.
2. Place some sample text files in the **"Reviews"** folder. 
3. Run the script to see language, sentiment, key phrases, entities, and links printed out for each file.

**Dependencies:** 
- Azure SDK: Required for Azure Text Analytics integration. 
- python-dotenv: To load environment variables.
